File: 67_add_binary/addBinary.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        k = len(a)
        m = len(b)
        n = max(k, m)

        a = a[::-1]
        b = b[::-1]

        res = ""
        carry = 0

        for i in range(n):
            valA = int(a[i]) if i < k else 0
            valB = int(b[i]) if i < m else 0

            total = valA + valB + carry
            val = total % 2

            res = str(val) + res
            carry = total // 2

        if carry:
            res = "1" + res
        return res

File: 67_add_binary/addBinary.ts
function addBinary(a: string, b: string): string {
  const k = a.length;
  const m = b.length;

  const n = Math.max(k, m);

  a = a.split("").reverse().join("");
  b = b.split("").reverse().join("");

  let res = "";

  let carry = 0;

  for (let i = 0; i < n; i++) {
    const valA = i < k ? +a[i] : 0;
    const valB = i < m ? +b[i] : 0;

    const total = valA + valB + carry;
    const val = (total % 2).toString();
    res = val + res;
    carry = Math.floor(total / 2);
  }

  if (carry > 0) {
    res = "1" + res;
  }

  return res;
}
